packages/api/cms-api/src/common/guards/header.guard.ts

```diff
@@ -0,0 +1,26 @@
+import { CanActivate, ExecutionContext, Injectable } from "@nestjs/common";
+import { GqlExecutionContext } from "@nestjs/graphql";
+import { Request } from "express";
+
+@Injectable()
+/** Blocks all requests without specified header */
+export class CometHeaderGuard implements CanActivate {
```

Comment:
HeaderGuard is unspecific, what about:

- BlockRequestsWithoutHeader (using your comment)
- BlockWithoutHeader

Reply:
BlockWithoutHeader sounds good and makes the comment useless :)